### PR TITLE
Shalev dev2

### PR DESCRIPTION
**Pull request by Shalev001**
all of the classes have been tested and function as expected

### CODE REVIEW

**Review by Fluff-E — APPROVED** [review]

File: FileProcessor/src/main/java/com/group7/FileProcessor/entries/LocFile.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public void updateContents(){
+        try {
+            file = new File(address);
+            
+            scanner = new Scanner(file);
+            
+            updateSize();// updating the stored size of the file
+            
+            //setting contents
+            String temp = "";
```
Comment (Tacitor) [inline FileProcessor/src/main/java/com/group7/FileProcessor/entries/LocFile.java:39]: Would you be able to use a StringBuilder for this?

File: FileProcessor/src/main/java/com/group7/FileProcessor/entries/LocFile.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            while (scanner.hasNext()) {
+                temp += scanner.nextLine() + "\n"; // running through all the lines in the file and adding them to temp
+            }
+            setContents(temp);
+
+        } catch (FileNotFoundException e) {
+            System.out.println("File Not Found, adress: " + address);
```
Comment (Tacitor) [inline FileProcessor/src/main/java/com/group7/FileProcessor/entries/LocFile.java:46]: Could you add printing the stack trace to the print statement too? Might help debugging later.

File: FileProcessor/src/main/java/com/group7/FileProcessor/entries/Local.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    /**
+     * a method to update the stored size of the file
+     */
+    public void updateSize() {
+        try {
+            //using built in nio.file.Files and nio.file.Paths to fill in length
+            Path path = Paths.get(getAddress());
+            setLength(Files.size(path));
+        } catch (IOException e) {
+            System.out.println("IOException in LocFile");
```
Comment (Tacitor) [inline FileProcessor/src/main/java/com/group7/FileProcessor/entries/Local.java:38]: Again if you don't mind if you could print out the stack trace here too.

**Review by Tacitor — CHANGES_REQUESTED** [review]
Please change the String to a StringBuilder in LocFile.java

Also if you don't mind adding stack traces to the catch statement print outs

**Review by Tacitor — APPROVED** [review]
Looks great!

**Pull request merged**